src/Raven.Client/Documents/Session/AsyncDocumentSession.cs

```diff
@@ -64,7 +66,28 @@ public async Task RefreshAsync<T>(T entity, CancellationToken token = default(Ca
                 var command = new GetDocumentsCommand(new[] { documentInfo.Id }, includes: null, metadataOnly: false);
                 await RequestExecutor.ExecuteAsync(command, Context, _sessionInfo, token).ConfigureAwait(false);
 
-                RefreshInternal(entity, command, documentInfo);
+                var commandResult = (BlittableJsonReaderObject)command.Result.Results[0];
+                RefreshInternal(entity, commandResult, documentInfo);
+            }
+        }
+
+        public async Task RefreshAsync<T>(T[] entities, CancellationToken token = default(CancellationToken))
+        {
+            DocumentInfo[] documentInfos = new DocumentInfo[entities.Length];
+            for (int i = 0; i < entities.Length; i++)
+            {
+                if (DocumentsByEntity.TryGetValue(entities[i], out documentInfos[i]) == false)
+                    throw new InvalidOperationException("Cannot refresh a transient instance");
+                IncrementRequestCount();
```

Comment:
you want to move this line out of the loop, at the end you are only sending 1 request, right?